winch/codegen/src/isa/aarch64/asm.rs

```diff
@@ -1284,3 +1288,81 @@ impl Assembler {
         });
     }
 }
+
+/// Captures the region in a MachBuffer where an add-with-immediate instruction would be emitted,
+/// but the immediate is not yet known.
+pub(crate) struct PatchableAddToReg {
+    /// The region to be patched in the [`MachBuffer`]. It contains
+    /// space for 3 32-bit instructions, i.e. it's 12 bytes long.
+    region: PatchRegion,
+
+    // The destination register for the add instruction.
+    reg: Reg,
+
+    // The temporary register used to hold the immediate value.
+    tmp: Reg,
+}
+
+impl PatchableAddToReg {
+    /// Create a new [`PatchableAddToReg`] by capturing a region in the output buffer where the
+    /// add-with-immediate sequence occurs. The [`MachBuffer`] will have and add-with-immediate instruction
```

Comment:
Perhaps we should update the documentation here, since the implementation here differs a bit from x64, e.g., we're emitting `add rd, rm, rn`

Reply:
Oh, ok. I already did alter it (note it describes an instruction _sequence_) but I can alter it to be a little more explicit.